zcash_client_backend/src/fees.rs

```diff
@@ -21,35 +21,61 @@ pub mod sapling;
 pub mod standard;
 pub mod zip317;
 
-/// A proposed change amount and output pool.
+/// `ChangeValue` represents either a proposed change output to a shielded pool
+/// (with an optional change memo), or if the "transparent-inputs" feature is
+/// enabled, an ephemeral output to the transparent pool.
 #[derive(Clone, Debug, PartialEq, Eq)]
-pub struct ChangeValue {
-    output_pool: PoolType,
-    value: NonNegativeAmount,
-    memo: Option<MemoBytes>,
+pub struct ChangeValue(ChangeValueInner);
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+enum ChangeValueInner {
+    Shielded {
+        protocol: ShieldedProtocol,
+        value: NonNegativeAmount,
+        memo: Option<MemoBytes>,
+    },
+    #[cfg(feature = "transparent-inputs")]
+    EphemeralTransparent { value: NonNegativeAmount },
 }
 
 impl ChangeValue {
-    /// Constructs a new change value from its constituent parts.
+    #[cfg_attr(
+        feature = "transparent-inputs",
+        doc = "Constructs a new change or ephemeral value from its constituent
+               parts. Currently, the only supported combinations are change sent
+               to a shielded pool, or an ephemeral output to the transparent pool."
+    )]
+    #[cfg_attr(
+        not(feature = "transparent-inputs"),
+        doc = "Constructs a new change value from its constituent parts.
+               Currently, `output_pool` must be a shielded pool (enable the
+               `transparent-inputs` feature to support ephemeral outputs to the
+               transparent pool). Use `ChangeValue::shielded` to avoid the
+               `Option` return."
+    )]
```

Comment:
I think that we should just remove `ChangeValue::new`; it has too much invalid state space in its arguments.

Reply:
Done.